File: src/main/java/eu/ibagroup/formainframe/utils/crudable/annotations/ForeignKey.java
/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright IBA Group 2020
 */

package eu.ibagroup.formainframe.utils.crudable.annotations;

import java.lang.annotation.*;

/**
 * Interface to describe the foreign key class assigned to the property
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
@Inherited
public @interface ForeignKey {
  Class<?> foreignClass();
}

File: src/main/kotlin/eu/ibagroup/formainframe/utils/crudable/annotations/ForeignKey.kt
/*
 * Copyright (c) 2020-2024 IBA Group.
 *
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   IBA Group
 *   Zowe Community
 */
package eu.ibagroup.formainframe.utils.crudable.annotations

import java.lang.annotation.Inherited
import kotlin.reflect.KClass

/**
 * Interface to describe the foreign key class assigned to the property
 */
@Target(AnnotationTarget.FIELD)
@Inherited
annotation class ForeignKey(val foreignClass: KClass<*>)
